content/ideas/algebraic-graphs.md

```diff
@@ -0,0 +1,62 @@
+---
+title: More graph algorithms for Alga
+---
+
+While graphs are ubiquitous in computer programming,
+they have long been underrepresented in Haskell
+due to the imperative nature and lack of type-safety
+of the existing implementations.
+
+The [Algebraic Graphs library](https://github.com/snowleopard/alga) has so far succeded 
+(thanks in part to last years summer of code) in providing a common, 
+typesafe interface for graph instances. However, general ecosystem adoption remains low,
+in part due to the lack of graph algorithms provided.
+
+Therefore the goal of this year's Summer of Code is
+to increase the usefulness of alga for a wider audience:
+
+## Goal 1: Develop a type-safe representation for acyclic graphs in alga.
+
+Acyclic graphs are both common and heavily used in dependency
+management. Improvements in this area would therefore directly
+benefit downstream packages like `build` or `aura`.
```

Comment:
You can also link to IOHK's Plutus project, which uses Alga: 

https://github.com/input-output-hk/plutus/blob/master/plutus-ir/src/Language/PlutusIR/Analysis/Dependencies.hs

Many of Alga's issues appear to come from there.

Reply:
I'd also add something like ", as well as a few commercial users of the library" -- I know for sure that a few companies are using Alga, but unfortunately can't go into any specifics, as this information is not public yet.